### DISCUSSION TITLE
Connecting to kernel: Python 3 (ipykernel): Waiting for Jupyter Session to be idle

### DISCUSSION THREAD

**Discussion by jaymc-arg** [microsoft/vscode-jupyter#15848] [Questions and Answers]
Hi there,
I'm trying to connect to a remote JupyterHub server from vscode, the extension detects de server, validates my user and pass and I select the kernel, but when I execute a cell or a line in a .py a notification pops up:

Connecting to kernel: Python 3 (ipykernel): Waiting for Jupyter Session to be idle

It hungs up there and never execute the selected code. I've already used the extension earlier and never happened before. I've already tried reinstalling the extension. Is this a bug?

jupyter extension: v2024.6.0
jupyterhub extension: v2023.10.1003012305
jupyterhub server version: 4.0.0

Thank's in advance.

**Comment by jsv1206** [comment]
I have a similar problem. Previously I hadn't used the jupyterHub extension, now VS code is forcing me to use this and it doesn't seem to work. 

Could you find a solution to this problem ?

**Reply by jaymc-arg** [reply]
There is still no solution, opened [this issue](https://github.com/microsoft/vscode-jupyter/issues/15852) (that a maintainer closed), but did not get it to work properly.

**Comment by seohyunjun** [comment]
I have same problem
[waiting  ~ inf]
![image](https://github.com/user-attachments/assets/ca13f7be-e96b-45b9-8357-6112a2c2a052)

```
05:17:53.412 ServerApp] Couldn't authenticate WebSocket connection
2024. 8. 29. 오후 2:17:53[I 2024-08-29 05:17:53.412 ServerApp] Setting new xsrf cookie for b':RyNY49sr-VtugeN1LgPCUizdq9EWSo=' {'path': '/user/hj/', 'max_age': 3600}
2024. 8. 29. 오후 2:17:53[W 2024-08-29 05:17:53.415 ServerApp] 403 GET /user/hj/api/kernels/51b8dc683af3132/channels?session_id=aae82&token=[secret] (@10667) 4.27ms
2024. 8. 29. 오후 2:17:55[W 2024-08-29 05:17:55.563 ServerApp] Couldn't authenticate WebSocket connection
2024. 8. 29. 오후 2:17:55[I 2024-08-29 05:17:55.564 ServerApp] Setting new xsrf cookie for b':RyNY49sr-VtugePY7ldq9EWSo=' {'path': '/user/hj/', 'max_age': 3600}
2024. 8. 29. 오후 2:17:55[W 2024-08-29 05:17:55.565 ServerApp] 403 GET /user/hj/api/kernels/51b8dc6ahfhff3132/channels?session_id=aa7b5ec7-4oken=[secret] (@) 3.60ms
2024. 8. 29. 오후 2:17:56[I 2024-08-29 05:17:56.425 JupyterHub log:192](B 200 GET /hub/api/user (hj@127.0.0.1) 4.48ms
2024. 8. 29. 오후 2:17:56[I 2024-08-29 05:17:56.426 ServerApp] 200 GET /user/hj/api/sessions?17276282 (hj) 9.63ms
2024. 8. 29. 오후 2:17:56[I 2024-08-29 05:17:56.522 ServerApp] 200 GET /user/hj/api/kernels?172420 (hj@0.27) 1.85ms
```

**Reply by jaymc-arg** [reply]
There is still no solution, opened [this issue](https://github.com/microsoft/vscode-jupyter/issues/15852) (that a maintainer closed), but did not get it to work properly.

**Reply by seohyunjun** [reply]
I hope it'll be work. 😢
